### PR TITLE
Make == less strict for links 

### LINKED ISSUE #376: Define a less strict == for link objects
```julia
julia> GLM.LogitLink() == GLM.LogitLink()
false
```

Probably not the intended behaviour 😄 

This also causes problems in MLJ, which wraps some GLM models. MLJ needs to be able to detect when model hyper-parameters change, when determining whether to retrain models (eg, in a pipeline with several components, some of which might change while others do not) -  and for other reasons.

### PR DESCRIPTION
**Pull request by ablaom**
Resolves #376

### CODE REVIEW

File: src/glmtools.jl [hunk 1]
```diff
@@ -79,20 +79,24 @@ mutable struct NegativeBinomialLink  <: Link
     θ::Float64
 end
 
+import Base.==
+==(lnk1::NegativeBinomialLink, lnk2::NegativeBinomialLink) =
```
Comment (nalimilan) [inline src/glmtools.jl:83]: Can't you just make `NegativeBinomialLink` immutable instead?

Also if you keep it:
```suggestion
Base.:(==)(lnk1::NegativeBinomialLink, lnk2::NegativeBinomialLink) =
```

Comment (ablaom) [reply src/glmtools.jl:83]: Sure.

File: test/runtests.jl [hunk 2]
```diff
@@ -694,3 +694,9 @@ end
         @test deviance(glm(@formula(y ~ x₁ + x₂), df, Binomial(), l, maxiter=40)) < 1e-6
     end
 end
+
+@testset "Issue #376 (== for links)" begin
+    @test GLM.LogitLink() == GLM.LogitLink()
+    @test NegativeBinomialLink(0.3) == NegativeBinomialLink(0.3)
+    @test NegativeBinomialLink(0.31) != NegativeBinomialLink(0.3)
+end
```
Comment (nalimilan) [inline test/runtests.jl:702]: While we're at it, better test the full series:

```suggestion
@testset "Issue #376 (== and isequal for links)" begin
    @test GLM.LogitLink() == GLM.LogitLink()
    @test NegativeBinomialLink(0.3) == NegativeBinomialLink(0.3)
    @test NegativeBinomialLink(0.31) != NegativeBinomialLink(0.3)

    @test isequal(GLM.LogitLink(), GLM.LogitLink())
    @test isequal(NegativeBinomialLink(0.3), NegativeBinomialLink(0.3))
    @test !isequal(NegativeBinomialLink(0.31), NegativeBinomialLink(0.3))

    @test hash(GLM.LogitLink()) == hash(GLM.LogitLink())
    @test hash(NegativeBinomialLink(0.3)) == hash(NegativeBinomialLink(0.3))
    @test hash(NegativeBinomialLink(0.31)) != hash(NegativeBinomialLink(0.3))
end
```

**Comment by ablaom** [comment]
@nalimilan Be good to get this merged and a patch out soon, thanks.

**Comment by ablaom** [comment]
@nalimilan @andreasnoack @ararslan Sorry to nag but this  simple PR was approved 27 days ago. Can we please have a merge and new minor release?

**Comment by ablaom** [comment]
@ararslan Many thanks ❤️

**Pull request merged**